### ISSUE TITLE
[body-changed-file-mention] not working when run as pre-commit hook

### ISSUE THREAD

**Issue by MaZZly** [jorisroovers/gitlint#105]
It seems that `commit.changed_files` is empty list when running gitlint as a [pre-commit](https://pre-commit.com/)-hook

### Setup
pre-commit is in use and hook installed with `pre-commit install --hook-type commit-msg`

`.gitlint`:
```
[body-changed-file-mention]
files=.gitlint
```

`.pre-commint-config.yaml`:
```
-   repo: https://github.com/jorisroovers/gitlint
    rev:  v0.12.0
    hooks:
    -   id: gitlint
```

### Reproduce:
1. Change `.gitlint` and commit, without mentioning .gitlint in body.
2. Run gitlint afterwards, it complains about modified file 
```
2: B7 Body does not mention changed file '.gitlint'
```
---

Debugging this, I created my own rule that will always reject, with list of modified files.
Doing commit with pre-commit:
```
gitlint..................................................................Failed
hookid: gitlint

2: UC1 Changed files: []
```

If I commit with `--no-verify` and then run `gitlint` It gives:
```
2: B7 Body does not mention changed file '.gitlint'
2: UC1 Changed files: [u'.gitlint']
```

Conclusion is that modified files list is empty when gitlint is run as a hook in pre-commit

**Comment by jorisroovers** [comment]
Good find, thanks for submitting the bug report :-)

There's 2 contributing factors here:

1. The integration between pre-commit and gitlint. When using pre-commit, gitlint does not fetch the commit message itself from the git repo, instead relying on pre-commit to pass the commit message via a file, using gitlint's ```--commit-msg``` flag.
2. When a commit message is passed via ```--commit-msg```, gitlint will not be querying the local git repository for additional information about the commit. This is because the last commit in the local repo might be a different commit from the one that is being linted via ```--commit-msg```.

This behavior will be the same when using gitlint's own commit-msg hook (```gitlint install-hook```) as it uses a similar mechanism under the hook (it passes the commit msg via stdin which causes gitlint to ignore local repo). I just verified that this is indeed the case.

What's really going on here is that gitlint currently doesn't really understand the concept of a pre-commit, instead relying on commit message passing to handle the pre-commit case. This is necessary as gitlint cannot read a non-committed message from the local repo.

I think the fix includes adding a ```--staged``` flag to tell gitlint to determine the set of staged files from the local repo and then use ```git diff --staged --name-only``` internally to determine the set of staged files.

Integrating this with pre-commit should just be a quick update to our [.pre-commit-hooks.yaml](https://github.com/jorisroovers/gitlint/blob/01addcdfc33f35995266ed75055dd361643c1e9d/.pre-commit-hooks.yaml) file.

I'll probably be able to work on this before the end of the year.

Meanwhile, you should be able to use this workaround with a user-defined rule:

```python
from gitlint.rules import CommitRule, RuleViolation
from gitlint.options import IntOption
from gitlint.git import _git

class CheckStagedFiles(CommitRule):
    name = "check-staged-files"
    id = "UC1"

    def validate(self, commit):
        print("body", commit.message.body) # string[]

        # NOTE: _git() is an internal, undocumented function. No guarantees on future availability or behavior :-)
        staged_files = _git("diff",  "--staged", "--name-only")

        # I'll leave the comparison logic to you :-)

        return [RuleViolation(self.id, "Staged file not mentioned in commit body", line_nr=1)]
```

Hope this helps!

**Comment by jorisroovers** [comment]
Released this as part of the 0.13.0 release yesterday but only added the `--staged` flag  to our `.pre-commit-hooks.yaml` file afterwards 🤦‍♂️. I'll be pushing 0.13.1 in next few hours to fix this.
